### PR TITLE
Bug fix for chef solo provisioner never validating

### PR DESCRIPTION
**Pull request by diamond29**
Make sure that an empty array does not become a required key.

### CODE REVIEW

File: lib/packer/provisioners/chef/solo.rb [hunk 1]
```diff
@@ -9,7 +9,7 @@ class Solo < Chef
         def initialize
           super
           self.data['type'] = CHEF_SOLO
-          self.add_required([])
+          #self.add_required([])
```
Comment (ianchesal) [inline lib/packer/provisioners/chef/solo.rb]: I think this is the only change we really need here, right? Don't push an empty array if there aren't any valid keys? The rest is probably unnecessary. I'm not a big fan of strong type checking ala `valid_key?` above -- I'd rather enforce as little as possible. It generally makes things more adaptable to uses you weren't anticipating.

File: spec/packer/provisioners/chef/solo_spec.rb [hunk 2]
```diff
@@ -95,4 +95,10 @@
       expect { provisioner.run_list(some_string) }.to raise_error
     end
   end
+
+  describe '#validate' do
+    it 'works with no data' do
```
Comment (ianchesal) [inline spec/packer/provisioners/chef/solo_spec.rb]: I like this new test!

File: lib/packer/provisioners/chef/solo.rb (hunk 1 above)
Comment (diamond29) [inline lib/packer/provisioners/chef/solo.rb]: That kind of logic makes sense to me, I think.

**Comment by diamond29** [comment]
Is this where I push follow up commits to my branch that removes the stuff you're suggesting I remove?

**Comment by ianchesal** [comment]
@diamond29 yea, you can push additional changes to the PR to make them. Would be nice to cut this down to just removing the push of the empty array and that one new spec test. The rest is unnecessary.

**Comment by diamond29** [comment]
Reverted the type checking stuff as specified - I think this can be merged/closed?

**Comment by ianchesal** [comment]
Thanks. I'll get it merged today.

**Pull request merged**